crates/transpiler/src/passes/disjoint_layout.rs

```diff
@@ -0,0 +1,545 @@
+                if matches!(op, StandardInstruction::Barrier(_)) {
+                    if let Some(label) = inst.label.as_ref() {
+                        if label.contains("_uuid=") {
+                            return Some(index);
+                        }
+                    }
+                }
+            }
+            None
+        })
+        .collect();
+    for node_index in barrier_nodes {
+        let num_qubits = dag[node_index].unwrap_operation().op.num_qubits();
+        let label = dag[node_index].unwrap_operation().label.clone().unwrap();
+        match uuid_map.get(label.as_str()) {
+            Some(other_index) => {
+                let num_qubits = dag[*other_index].unwrap_operation().op.num_qubits() + num_qubits;
+                let new_label = if retain_uuid {
+                    Some(label.to_string())
+                } else if label.starts_with("_none_uuid=") {
+                    None
+                } else {
+                    let len = label.len();
+                    let components: Vec<&str> = label.split("_uuid=").collect();
+                    Some(components[..len - 1].join("_uuid="))
+                };
+                let new_op = PackedOperation::from_standard_instruction(
+                    StandardInstruction::Barrier(num_qubits),
+                );
+                let new_node = dag.replace_block(
+                    &[*other_index, node_index],
+                    new_op,
+                    SmallVec::new(),
+                    new_label.as_deref(),
+                    true,
+                    &HashMap::new(),
+                    &HashMap::new(),
+                )?;
+                uuid_map.insert(*label, new_node);
+            }
+            None => {
+                uuid_map.insert(*label, node_index);
+            }
+        }
+    }
+    Ok(())
+}
+
+fn split_barriers(dag: &mut DAGCircuit) -> PyResult<()> {
+    for (_index, inst) in dag.op_nodes(true) {
+        if let OperationRef::StandardInstruction(StandardInstruction::Barrier(num_qubits)) =
+            inst.op.view()
+        {
+            if num_qubits == 1 {
+                continue;
+            }
+            let barrier_uuid = match &inst.label {
+                Some(label) => format!("{}_uuid={}", label, Uuid::new_v4()),
+                None => format!("_none_uuid={}", Uuid::new_v4()),
+            };
+            let mut split_dag = DAGCircuit::new()?;
+            for q in 0..num_qubits {
+                split_dag.add_qubit_unchecked(ShareableQubit::new_anonymous())?;
+                split_dag.apply_operation_back(
+                    PackedOperation::from_standard_instruction(StandardInstruction::Barrier(1)),
+                    &[Qubit(q)],
+                    &[],
+                    None,
+                    Some(barrier_uuid.clone()),
+                    #[cfg(feature = "cache_pygates")]
+                    None,
+                )?;
+            }
+        }
+    }
```

Comment:
One day I'll convince you to use `let ... = ... else { continue; };` instead of indenting the whole block twice haha.

Reply:
This is what happens when I learn a language too early. I don't have newer language constructs in my muscle memory :) Let else was added in 1.65 which is too new, I'll get used to using it at some point.

Reply:
https://github.com/Qiskit/qiskit/pull/14177/commits/18a38752a6bc547b38de71ab84e9d473c89740ea